src/meta/src/stream/source_manager.rs

```diff
@@ -604,15 +604,21 @@ where
                 tracing::warn!("Failed to unregister_table_ids {:#?}.\nThey will be cleaned up on node restart.\n{:#?}", registered_table_ids, e);
             }
         }));
-        let futures = self.all_stream_clients().await?.into_iter().map(|client| {
-            let request = ComputeNodeCreateSourceRequest {
-                source: Some(source.clone()),
-            };
-            async move { client.create_source(request).await }
-        });
 
-        // ignore response body, always none
-        let _ = try_join_all(futures).await?;
+        match source.info.as_ref().unwrap() {
+            Info::TableSource(_) => {
+                let futures = self.all_stream_clients().await?.into_iter().map(|client| {
+                    let request = ComputeNodeCreateSourceRequest {
+                        source: Some(source.clone()),
```

Comment:
Actually, we can also carry the table source info just like the stream source, and lazily register it on the compute node's (table) source manager so that this RPC can be totally eliminated. Let's do this in the next PRs.

Reply:
Yes~I will do this refactor in next pr. Maybe after National Day Holiday : )